sovrin_client/agent/agent.py

```diff
@@ -202,12 +217,73 @@ def _initIssuerProverVerifier(self):
         self.prover = SovrinProver(client=self.client, wallet=self._wallet)
         self.verifier = SovrinVerifier(client=self.client, wallet=self._wallet)
 
+    @property
+    def wallet(self):
+        return self._wallet
+
+    @wallet.setter
+    def wallet(self, newWallet):
+        self._wallet = newWallet
+
     @Agent.client.setter
     def client(self, client):
         Agent.client.fset(self, client)
         if self.client:
             self._initIssuerProverVerifier()
 
+    def getContextDir(self):
+        # TODO: Will we need any environment context for agent's walletS?
+        return os.path.expanduser(os.path.join(
+            self.config.baseDir, self.config.keyringsDir, "agents",
+            self.name.lower().replace(" ", "-")))
+
+    def start(self, loop):
+        super().start(loop)
+        self._restoreLastActiveWallet()
+
+    def _saveAllWallets(self):
```

Comment:
Well, the idea was that there will be different wallets in an agent and we'll have to save different wallets, so for now, it may seem unnecessary, but I can add a TODO probably to convey the intention that there are some other wallets which needs to be persisted (like inmemorywallets etc).